lib/slack_announcer.rb

```diff
@@ -7,22 +7,33 @@ def initialize(environment_name, slack_url)
   end
 
   def announce_start(repo_name, application, slack_channel = "#govuk-deploy")
+    return unless staging_or_production?
+
     text = "#{environment_emoji} :mega: #{version_and_link(repo_name, application)} " \
            "is being deployed to *#{@environment_name}* by #{build_user}"
 
     post_text(slack_channel, text)
   end
 
   def announce_done(repo_name, application, slack_channel = "#govuk-deploy")
+    return unless staging_or_production?
+
     text = "#{environment_emoji} :white_check_mark: #{version_and_link(repo_name, application)} " \
            "deployed to *#{@environment_name}* by #{build_user}"
 
     post_text(slack_channel, text)
   end
 
-  def post_text(slack_channel, text)
-    return unless %w[production staging].include?(@environment_name)
+  def announce_failed(repo_name, application, slack_channel = "#govuk-deploy")
+    return unless staging_or_production?
```

Comment:
We probably want to announce a failed deploy even if it's on integration.

Reply:
I'm not sure that we do, because lots of people deploy their own branches to integration which is reasonable for them to fail - and that might mean we start to ignore those messages.

Reply:
Hmm, fair enough. I think it's very rare for a deployment to fail even on integration, but good point about the noise (although at least it wouldn't be spamming govuk-2ndline).

Reply:
Currently we don't announce anything about Integration (started or succeeded) so people won't be looking for failed messages either